account-kit/infra/src/middleware/gasManager.ts

```diff
@@ -26,3 +45,159 @@ export function alchemyGasManagerMiddleware(
     context: { policyId: policyId },
   });
 }
+
+/**
+ * Paymaster middleware factory that uses Alchemy's Gas Manager for sponsoring transactions using
+ * the `alchemy_requestGasAndPaymasterAndData` method instead of standard ERC-7677 methods.
+ *
+ * @example
+ *  ```ts
+ * import { sepolia, alchemyGasAndPaymasterAndDataMiddleware } from "@account-kit/infra";
+ * import { http } from "viem";
+ *
+ * const client = createSmartAccountClient({
+ *  transport: http("rpc-url"),
+ *  chain: sepolia,
+ *  ...alchemyGasAndPaymasterAndDataMiddleware("policyId")
+ * });
+ * ```
+ *
+ * @param {string} policyId the policyId for Alchemy's gas manager
+ * @returns {Pick<ClientMiddlewareConfig, "dummyPaymasterAndData" | "paymasterAndData">} partial client middleware configuration containing `dummyPaymasterAndData` and `paymasterAndData`
+ */
+export function alchemyGasAndPaymasterAndDataMiddleware(
+  policyId: string
+): Pick<
+  ClientMiddlewareConfig,
+  "dummyPaymasterAndData" | "feeEstimator" | "gasEstimator" | "paymasterAndData"
+> {
+  return {
+    dummyPaymasterAndData: async (uo, { client, account }) => {
+      if (!client.chain) {
+        throw new ChainNotFoundError();
+      }
+
+      const paymaster = getAlchemyPaymasterAddress(client.chain);
+      const paymasterData = await account.getDummySignature();
+
+      return {
+        ...uo,
+        dummyPaymasterAndData: {
+          paymaster,
+          paymasterData,
+        },
+      };
+    },
+    feeEstimator: noopMiddleware,
```

Comment:
do we really want to override here? 

https://github.com/alchemyplatform/aa-sdk/pull/840/files#diff-f26c35f21daf3d855358970d4b5c9bd096ffb135ecdd73f2b67b5d67351f3261L202-L247

Reply:
following up in slack.

thanks for sharing that diff btw... i pushed a couple of changes based on reviewing that.